### PR TITLE
Explain random seeds in episode 2

### LINKED ISSUE #489: Explain why we use random seeds
In episode 2 it might be worth emphasising when and when not to use random seeds, in case learners simply "copy" what they learnt here?

### PR DESCRIPTION
**Pull request by svenvanderburg**
Fixes #489

### CODE REVIEW

File: episodes/2-keras.Rmd [hunk 1]
```diff
@@ -283,6 +283,14 @@ from sklearn.model_selection import train_test_split
 X_train, X_test, y_train, y_test = train_test_split(features, target, test_size=0.2, random_state=0, shuffle=True, stratify=target)
 ```
 
+::: callout
+## Importance of using the same train test split
+By setting `random_state==0` we ensure that everyone has the same train-test split.
```
Comment (carschno) [inline episodes/2-keras.Rmd:288]:
```suggestion
By setting `random_state=0` we ensure that everyone has the same train-test split.
```

File: episodes/2-keras.Rmd [hunk 2]
```diff
@@ -283,6 +283,14 @@ from sklearn.model_selection import train_test_split
 X_train, X_test, y_train, y_test = train_test_split(features, target, test_size=0.2, random_state=0, shuffle=True, stratify=target)
 ```
 
+::: callout
+## Importance of using the same train test split
```
Comment (carschno) [inline episodes/2-keras.Rmd:287]: For consistency:
```suggestion
## Importance of using the same train-test split
```

File: episodes/2-keras.Rmd [hunk 3]
```diff
@@ -312,6 +320,16 @@ seed(1)
...
 
+::: callout
+## When to use random seeds?
+We use a random seed here to ensure that we get the same results every time we run this code.
+This makes our results reproducible and allows us to better compare results between different experiments.
+
+Please note that even though you have selected a random seed, this seed is used to generate a
+**different** random number every time you execute a Jupyter cell.
+So, to get truly replicable deep learning pipelines you need to run the notebook from start to end in one go.
+:::
```
Comment (carschno) [inline episodes/2-keras.Rmd:331]: I have seen some confusion about this in the past, so perhaps we can make it even more explicit:
```suggestion
When the same code is run on another machine, the random number and hence the results can also differ.
:::
```

**Review by carschno — COMMENTED** [review]
I've had some minor comments.

File: episodes/2-keras.Rmd (hunk 3 above)
Comment (svenvanderburg) [reply episodes/2-keras.Rmd:331]: @carschno I think this is actually not true. Or at least not the promise of using random seeds right? The whole idea of using random seeds is that the code becomes deterministic and the results are the same on another machine. Of course you should always double check that.

I agree that often students have a different result, but that is probably due to running cells multiple times or different dependency versions.

Comment (svenvanderburg) [reply episodes/2-keras.Rmd:331]: As discussed with @carschno we decided to leave it like was proposed in this PR.
